src/EStopManager.cpp

```diff
@@ -0,0 +1,92 @@
+#include <EStopManager.h>
+#include "Arduino.h"
+
+#include "VisualStateManager.h"
+
+using namespace OpenShock;
+
+static EStopManager::EStopStatus_t s_estopStatus = EStopManager::EStopStatus_t::ALL_CLEAR;
+static char s_estopName[20];
+static unsigned long s_estopHoldToClearTime = 5000;
+static unsigned long s_lastEStopButtonStateChange = 0;
+static unsigned long s_estoppedAt = 0;
+static bool s_lastEStopButtonState = HIGH;
+
+static std::uint8_t s_estopPin;
+
+// TODO?: Allow active HIGH EStops?
+void EStopManager::Init() {
+    #ifdef OPENSHOCK_ESTOP_PIN
+    s_estopPin = OPENSHOCK_ESTOP_PIN;
+    pinMode(s_estopPin, INPUT_PULLUP);
+    snprintf(s_estopName, sizeof(s_estopName), "EStop-%u", s_estopPin);
```

Comment:
So, my reason to do this in other parts of the source code is becuase the classes there are instanciated, there can be more than one pin pattern manager for the same source file since its a very generic class that can have many use cases.
This is static so I would keep it to fixed logging tags. :p

Reply:
Fair, but this also does have to do with a specific GPIO pin, so keeping the consistency of reporting the pin with the corresponding manager would be nice.

What would you recommend?

Reply:
Probably just make it like I have in the other static files.
Before the ``using OpenShock;`` statement just put ``const char* const TAG = "Something";``
I need to revert this in the other files because these tags will be used to filter log output, they should never be variable 😅

Reply:
Fair, I guess. But there should still be a good way for us to see the user's pin assignments through logs. And with chips like the ESP32-S3 with onboard USB Serial, it's really hard to catch the first bit of Serial data as the COM port isn't available until after the device is powered and initializing.

Reply:
Set to static TAG.